src/serac/physics/boundary_conditions/boundary_condition.cpp

```diff
@@ -118,33 +118,45 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat, mfem::Vector& rhs, mf
 const mfem::Coefficient& BoundaryCondition::scalarCoefficient() const
 {
   auto scalar_coef = get_if<std::shared_ptr<mfem::Coefficient>>(&coef_);
-  SLIC_ERROR_ROOT_IF(!scalar_coef,
-                     "Asking for a scalar coefficient on a BoundaryCondition that contains a vector coefficient.");
-  return **scalar_coef;
+  if (scalar_coef) {
+    return **scalar_coef;
+  } else {
+    SLIC_ERROR_ROOT("Asking for a scalar coefficient on a BoundaryCondition that contains a vector coefficient.");
+    exit(1);
```

Comment:
SLIC_ERROR_ROOT handles this

Reply:
Jamie explained its the compiler warning there isn't a return/exit from this case. I'll log a SLIC bug.